### PR TITLE
[FEATURE] JSON output format

### PR DESCRIPTION
**Pull request by HorlogeSkynet**
```bash
python3 -m archey --json
```

## Description

~~Simple JSON-formatting of entries data.
It's a first basic support, maybe it could be enhanced in the future.~~

**EDIT 2020-05-13** : To keep things clean, DRY and concerns-separated, @ingrinder and myself started to work on a slight _rework_ to make entries' values independent from their corresponding (regular and original) text output.
This will _drastically_ simplify future work on entries data.

## Reason and / or context

It would allow one to use Archey as an API or so :smile:

## How has this been tested ?

Test cases and our machines.

## Types of changes :

- Bug fix (non-breaking change which fixes an issue)
- Typo / style fix (non-breaking change which improves readability)
- [X] New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist :

- [X] \[IF NEEDED\] I have updated the test cases (which pass) accordingly ;
- [X] My changes looks good ;
- [X] I agree that my code may be modified in the future ;
- [X] My code follows the code style of this project ([PEP8](https://www.python.org/dev/peps/pep-0008/)).

### CODE REVIEW

**Comment by ingrinder** [comment]
Huh, it's funny the CI build passed, since I'm consistently getting a unit test error on this branch:
```pytb
======================================================================
ERROR: test_json_output_format (archey.test.test_archey_output.TestOutputUtil)
Test how the `output` method handles JSON preferred formatting of entries
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/mock.py", line 1348, in patched
    return func(*newargs, **newkeywargs)
  File "/home/michael/git/archey4/archey/test/test_archey_output.py", line 504, in test_json_output_format
    output = Output(format_to_json=True)
  File "/home/michael/git/archey4/archey/output.py", line 59, in __init__
    if ansi_color and Configuration().get('colors_palette')['honor_ansi_color']:
KeyError: 'honor_ansi_color'

----------------------------------------------------------------------
Ran 108 tests in 0.075s

FAILED (errors=1)
```

...whether I use `python -m unittest` or `python setup.py -q test`. I haven't got a custom configuration file in `/etc/archey4` or `~/.config/archey4`...

**Comment by HorlogeSkynet** [comment]
Yes sorry, related to `d559edc52d839654989f1b39e6b0d3159bb71497`.
It's (again) related to how `Configuration` is mocked across test modules (and the fact that it is actually a Singleton never getting reset)...
It should be fixed on your machine now :bow:

**Comment by HorlogeSkynet** [comment]
'should be well cleaner now.
I'll even fix a little blemish in our `Output.{append,output}` flow when #71 will be merged.
Tell me what you think about that !

Bye :wave:

**Comment by ingrinder** [comment]
Sorry for the delays, I've been a little busy 😃. I made a few more modifications since I wasn't fully satisfied after I started work on e709d0e until seeing 32cdee5 inspired me to change how we treat `Entry.value`. Anyway, I'll try and get back to you with a more detailed response tomorrow if I get the chance 👍.

**Comment by HorlogeSkynet** [comment]
Here we are ! Thanks for your huge work on improving the (raw) value/text output integration.
I think it would allow more modifiability in the future, currently costing us time though, sorry for that.

Once this has been merged, I'll rebase #71 & #72, leaving you #67 :wink:

Anyway, there are eventually _many_ changes, so a proper review is needed :fearful:

**Comment by HorlogeSkynet** [comment]
Unfortunately a rebase against new `master` was a PITA, so I've simply merged it to resolve new conflicts.
Due to this PRs merging order, I've had to roll-back chunks of #71 not required anymore according to global rework/cleanup done here.

Anyway, a proper review is more than required, but future looks simpler once this patch will be merged :+1:

File: archey/entries/temperature.py [hunk 1]
```diff
@@ -41,23 +40,16 @@ def __init__(self, *args, **kwargs):
...
-        # Final average computation.
-        self.value = '{0}{1}{2}'.format(
-            str(round(sum(self._temps) / len(self._temps), 1)),
-            char_before_unit,
-            'F' if use_fahrenheit else 'C'
-        )
+        # Final average and maximum computations.
+        self.value = {
+            'temperature': float(round(sum(self._temps) / len(self._temps), 1)),
+            'max_temperature': float(round(max(self._temps), 1)),
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/entries/temperature.py:48]: Yes desired.
The goal is to simplify any final usage by the developer.
If their wish is to retrieve the maximum value, let's not try to hide it from them due to others.

File: archey/entries/uptime.py [hunk 2]
```diff
@@ -23,34 +23,11 @@ def __init__(self, *args, **kwargs):
...
-                uptime += 's'
-        elif not days and not hours:
-            uptime = '< 1 minute'
-
-        self.value = uptime
+        self.value = {
+            'days': days,
+            'hours': hours,
+            'minutes': minutes,
+        }
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/entries/uptime.py:30]: Let's keep this simple : As `uptime` gets down to the second on some systems, we wouldn't really take benefits from serializing the whole object...
If we implement a custom serializer, this means developers will have to implement a custom deserializer in their language too, and this is not desirable.

Although, I've added the seconds to the final API data to increase precision :+1:

File: archey/test/test_archey_api.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            }
+        )
+
+        # Meta-data verifications.
+        self.assertIn('meta', output_json_document)
+        # Check the SemVer segments types (should be integers).
+        for semver_segment in output_json_document['meta']['version']:
+            self.assertTrue(isinstance(semver_segment, int))
+        # Check that generated `date` meta-data is correct and not in the future.
+        self.assertGreater(
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/test/test_archey_api.py:63]: You're right :+1:

File: archey/__main__.py [hunk 4]
```diff
@@ -63,19 +63,28 @@ class Entries(Enum):
...
+        help='output entries data to JSON format'
+    )
     parser.add_argument(
         '-v', '--version',
-        action='version', version=__version__
+        action='version',
+        version=__version__
     )
-    parser.parse_args()
+    args = parser.parse_args()
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/__main__.py:76]: Yes, foreseen :wink:

File: archey/test/test_archey_gpu.py [hunk 5]
```diff
@@ -137,12 +145,16 @@ def test_no_match(self, _, __):
...
-        output = []
-        gpu_entry.output(output)
+        gpu = GPU()
+
+        output_mock = MagicMock()
+        gpu.output(output_mock)
 
-        self.assertFalse(gpu_entry.gpu_devices)
-        self.assertFalse(output[0][1])
+        self.assertFalse(gpu.value)
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/test/test_archey_gpu.py:153]: I've tried something, tell me what you think about it :wink:

File: archey/test/test_archey_colors.py [hunk 6]
```diff
@@ -51,6 +51,14 @@ def test_ansi_ecma_regexp(self):
...
         )
 
+    def test_remove_colors(self):
+        """Test our ANSI/ECMA REGEXP colors removal method"""
+        self.assertFalse(Colors.remove_colors(str(Colors.CLEAR)))
+        self.assertEqual(
+            Colors.remove_colors('\x1b[0;31mTEST\x1b[0;0m'),
+            'TEST'
+        )
+
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/test/test_archey_colors.py:61]: Done.

File: archey/entries/ram.py [hunk 7]
```diff
@@ -46,6 +46,21 @@ def __init__(self, *args, **kwargs):
             if used < 0:
                 used = total - ram['MemFree']
 
+        self.value = {
+            'used': used,
+            'total': total,
+            'unit': 'MiB'
+        }
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/entries/ram.py:53]: I think the idea _was_ to leave a 10^6 unit for `RAM` as most of systems do not exceed 10^9 (for `Disk`, that's not the case).
This predicate can be discussed, and may be easily changed in the future :+1:

File: archey/output.py [hunk 8]
```diff
@@ -68,27 +69,45 @@ def add_entry(self, module):
...
+                entry.output(self)
+            self._output_text()
 
+    def _output_json(self):
+        """
+        Finally outputs entries data to JSON format.
+        See `archey.api.JSONAPI` for further documentation.
+        """
+        print(
+            API(self._entries).json_serialization(pretty_print=True)
```
[the remark this answers is not in the document]
Comment (HorlogeSkynet) [reply archey/output.py:103]: Done ! I've opted for a simpler solution with only one flag.

**Pull request merged**